Code/GraphMol/Chirality.cpp

```diff
@@ -3072,7 +3073,7 @@ void setDoubleBondNeighborDirections(ROMol &mol, const Conformer *conf) {
   bool resetRings = false;
   if (!mol.getRingInfo()->isInitialized()) {
     resetRings = true;
-    MolOps::fastFindRings(mol);
```

Comment:
The ring finding is potential error prone because you cannot tell which type of ring-finding as been done.  This really should be the SSSR symm call - a few lines lower it calles isBondCandidateForStereo,  which calls shouldDetectDoubleBondStereo.  This in turn tests  for "ri->minBondRingSize(bond->getIdx()) >=
              Chirality::minRingSizeForDoubleBondStereo)"
              
              I have a test in which the minBondRingSize returns 10 if the fastFindRings is used, and this allows a bond in a 6 membered ring to proceed with stereo checking/assignment.  Now as it turns out, if sanititze is not on, that bond is not set to a stereo value anyway.  If sanitize or removeHs is on, the rings are recalculated as SSSR-symm anyway, and after some processing, the bond is NOT set to stereo.

So in the end, I do not have a test that fails if the wrong ring finding is done here, but it is calling a lot more code before deciding NOT to set the stereo.

In the next PR (atropisomers) I have added an attribure to the ring information class that indicates what find of ring finding was done, so that the calls to SSSR-SYmm can be done 1 and only 1 time if needed.  This is potentially more effecient and much safer.  

I can/could work that into this PR if you like - I think I probably should.

Reply:
I'm a little late to the party, but I just filed https://github.com/rdkit/rdkit/issues/6951 and almost immediately closed it again (I found it in the 2023.09.1 release, but noticed it was fixed in master), which was triggered by the issue @tadhurst-cdd described here :)